caput/misc/profiling.py

```diff
@@ -0,0 +1,137 @@
+"""Profiling for the caput pipeline"""
```

Comment:
How about just puttomg this under a `caput.profiling` module, rather than making `caput.misc` a subpackage?

Reply:
Even before this addition, I think the caput package structure should rather get sorted into fewer sub-packages instead of adding more modules.

Reply:
Do you have a proposal as to how you would organise it? My initial reaction is that it only makes sense if you can group related things together and I really don't see what they'd be. At a module level things are currently logically grouped together which seems reasonable. 

I really don't see how `caput.misc.profiling` makes much sense though. It's not like anything else is more organised. I think we should probably just put this `caput.profiling` and consider a reorganisation later on, although backwards compatibility will make it difficult to do anything too major without lots of work.

Reply:
> Do you have a proposal as to how you would organise it? My initial reaction is that it only makes sense if you can group related things together and I really don't see what they'd be. At a module level things are currently logically grouped together which seems reasonable.
Right, but we can make subpackages out of most modules and split those into multiple files.

Reply:
Yeah, I'd agree with that for the big ones like `memh5` and `pipeline`, but I'm not sure if any of the others are large enough. I'd still put the profiling stuff in its own root-level module.

Reply:
I don't think `profiling` or `cache` are relevant enough to be a top level submodule.